File: projects/10/Owl.java
/* Owl.java provides an Owl subclass of Bird.
 * 
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Daniela Contreras
 * Date: 03/30/20222
 ******************************************************/

public class Owl extends FlyingBird {

 /* explicit constructor
  * Receive: name, a String 
  * PostCond: myName == name.
  */
  public Owl()
  {
    super();
  }

  public Owl(String name)
  {     
    super(name);
  } 

 /* A Goose's Call
  * Return: a goose-call ("Honk!").
  */

  public String call()
  {
    return "Hoot!";
  } 
} 

File: projects/10/ruby/Owl.rb

# Owl.rb | Defines the Owl class which inherits attributes and methods
#   from the FlyingBird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Daniela Contreras
# Date: 03/25/22
####################################################

require './FlyingBird.rb'

class Owl < FlyingBird

##############################################
## Call returns sound of Owl                 #
##Return: Whoot! a string                    #
##############################################
    def call
        'Who-hoot!'
    end
    
end